faiss_vector_posting.go

```diff
@@ -403,3 +403,20 @@ func (sb *SegmentBase) InterpretVectorIndex(field string) (segment.VectorIndex,
 	vecIndex, err = faiss.ReadIndexFromBuffer(indexBytes, faiss.IOFlagReadOnly)
 	return wrapVecIndex, err
 }
+
+func (sb *SegmentBase) UpdateFieldStats(stats segment.FieldStats) {
+	for _, fieldName := range sb.fieldsInv {
+		pos := int(sb.fieldsSectionsMap[sb.fieldsMap[fieldName]-1][SectionFaissVectorIndex])
```

Comment:
you can avoid the sb.fieldsMap[fieldName] over here, since I believe the index of fieldsInv is the fieldID. you can verify it and do the change.

Reply:
I'm keeping this the same because there seems to be one case where the fieldsInv looks like this

_id _all content id _id _all content id

So basically theres copies of stuff. I'm not sure where the copies are coming from but for safety reasons (The process will crash with index out of range error otherwise) I'm keeping it in.